src/image.c

```diff
@@ -710,13 +708,7 @@ static void key_sha384(uint8_t key_slot, uint8_t *hash)
         return;
 
     wc_InitSha384(&sha384_ctx);
-    while (i < (uint32_t)(pubkey_sz)) {
```

Comment:
Why are you removing this logic? Guessing it is not required since the public key buffer is fully available? Is it possible that WOLFBOOT_SHA_BLOCK_SIZE might be useful for other reasons?

Reply:
wolfCrypts internal logic already loops over in the correct block size increments, the keybuffer is fully available so it's just wasted code size

Reply:
This is a different block size value. This is used to break the SHA processing into smaller chunks. This is typically used when the value being hashed comes from flash. This would read (by default) 128 bytes at a time.

Reply:
good to know, then it's almost certainly being re-adjusted by
```
        if ((i + blksz) > (uint32_t)pubkey_sz)
            blksz = pubkey_sz - i;
```
for ed and ecc keys, maybe rsa has bigger keys but in any case the entire key is still available and `wc_Sha384Update` can handle it all in a single function call

Reply:
I agree and I am okay removing this logic. Please make sure you do it consistently for the other algorithms too.

Reply:
No fun! I wanted to make an intern do that